ipatests/test_integration/test_ipalib_util.py

```diff
@@ -0,0 +1,37 @@
+# Authors:
+#   Felipe Barreto Volpone <fbarreto@redhat.com>
+#
+# Copyright (C) 2017  Red Hat
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Test the `ipalib.util` module.
+"""
+
+from ipalib import util
+from ipaplatform.paths import paths
+
+import pytest
+pytestmark = pytest.mark.tier0
+
+
+def test_imported_paths():
+    """
+    Test if constants created in ipalib.util are in sync with
+    paths declared in ipaplatform.paths
+    """
+    assert util.IPA_CLIENT_SYSRESTORE == paths.IPA_CLIENT_SYSRESTORE
+    assert util.IPA_DEFAULT_CONF == paths.IPA_DEFAULT_CONF
```

Comment:
This test will never fail, even if the hard-coded constants ``ipalib/util.py`` and ipaplatforms come out of sync. Since ``ipalib.util`` first tries to import from ``ipaplatform`` and ``ipaplatform`` is available, you are testing that ``paths.IPA_CLIENT_SYSRESTORE == paths.IPA_CLIENT_SYSRESTORE``.

Reply:
I've tried some different things, like:
```python
# test_ipalib_util.py
from ipaplatform.paths import paths
import sys
sys.modules['ipaplatform'] = None
from ipalib import util
import pytest
```

but in `ipalib/util.py`,
```python
try:
    import sys
    print(sys.module['ipaplatform'] is not None)  # returns True
    # pylint: disable=import-error,ipa-forbidden-import
    from ipaplatform.paths import paths
    # pylint: enable=import-error,ipa-forbidden-import
    IPA_CLIENT_SYSRESTORE = paths.IPA_CLIENT_SYSRESTORE + "1"
    IPA_DEFAULT_CONF = paths.IPA_DEFAULT_CONF

except ImportError:
    IPA_CLIENT_SYSRESTORE = "/var/lib/ipa-client/sysrestore1"
    IPA_DEFAULT_CONF = "/etc/ipa/default.conf"
``` 

`sys.module['ipaplatform'] is not None` returns True.

Or 

```python
import sys
ipalib = sys.modules['ipalib']
sys.modules['ipalib'] = None
from ipaplatform.paths import paths
sys.modules['ipalib'] = ipalib
sys.modules['ipaplatform'] = None
sys.modules['ipaplatform.paths'] = None
from ipalib import util
```
I've also tried some hacks using pytest, but they didn't work. Any idea?